src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1189,19 +1191,28 @@ public Component getTableCellRendererComponent(JTable table, Object value,
                 Icon icon = chooser.getIcon(file);
                 setIcon(icon);
 
-            } else if (value instanceof Long) {
-                long len = ((Long) value) / 1024L;
+            } else if (value instanceof Long len) {
                 if (listViewWindowsStyle) {
-                    text = MessageFormat.format(kiloByteString, len + 1);
+                    if(len == 0) {
+                        text = MessageFormat.format(kiloByteString, len);
+                    } else {
+                        len /= 1024L;
+                        text = MessageFormat.format(kiloByteString, len + 1);
+                    }
                 } else if (len < 1024L) {
-                    text = MessageFormat.format(kiloByteString, (len == 0L) ? 1L : len);
+                    text = MessageFormat.format(kiloByteString, (len==0 ? 0L : 1L));
                 } else {
                     len /= 1024L;
                     if (len < 1024L) {
-                        text = MessageFormat.format(megaByteString, len);
+                        text = MessageFormat.format(kiloByteString, len);
```

Comment:
So ALL of these have an "off by one" issue don't they ?

Here a file that is (for example) 2047 bytes in size will be displayed as "1KB"

Reply:
I think this also deserves extracting it into a separate function, and a unit test.
And perhaps another unit test with a non-English locale.

Reply:
> So ALL of these have an "off by one" issue don't they ?
> 
> Here a file that is (for example) 2047 bytes in size will be displayed as "1KB"

There is "off by one" issue in this case. One possible solution can be - 


![FileChooser_FileSizes](https://user-images.githubusercontent.com/107542245/179674833-345a7e81-4af2-403e-b9e1-9d690bffeddd.png)

Reply:
If any file having size  >1024 byte & <2048 byte will be rounded off as 2 KB and so on.
Similarly if file size is >1023KB will show file size as 1MB (rounding off will make file size as 1024KB and then it will convert it to MB).

Reply:
![TestFileSize](https://user-images.githubusercontent.com/107542245/179677108-3ef9784e-fa28-464c-8b09-5a940dd0f57a.png)

File size shown by JFileChooser in details view.